src/main/java/core/option/UserColorsPanel.java

```diff
@@ -0,0 +1,340 @@
+		/**
+		 * 0: The color name column is not editable
+		 * 1: The color reference name column is editable
+		 * 2: The color value column is only editable if reference is not set
+		 * 3: The default color column is only editable if set
+		 * @param row             the row whose value is to be queried
+		 * @param column          the column whose value is to be queried
+		 * @return true if cell is editable
+		 */
+		@Override
+		public boolean isCellEditable(int row, int column) {
+			return switch (column) {
+				case 1 -> true; // colorReference is editable
+				case 2 -> {    // color is only editable if colorReference is not set
+					var colorReference = this.getDataVector().get(row).get(1);
+					yield colorReference == null;
+				}
+				case 3 -> this.getDataVector().get(row).get(3) != null; // default reset is only possible if not null
+				default -> false; // others are not editable
+			};
+		}
+	};
+	private final JTable colorTable = new JTable(tableModel);
+
+	private final String[] columnNames = new String[]{
+			HOVerwaltung.instance().getLanguageString("Name"),
+			HOVerwaltung.instance().getLanguageString("Reference"),
+			HOVerwaltung.instance().getLanguageString("Value"),
+			HOVerwaltung.instance().getLanguageString("Default"),
+	};
+	private JPanel tablePanel;
+	private List<HOColor> colors = new ArrayList<>();
+
+	/**
+	 * Create user color setting panel
+	 */
+	protected UserColorsPanel() {
+		initComponents();
+	}
+
+	/**
+	 * Create label component with colored background, used to render the color columns
+	 * @param table Table
+	 * @param value Value
+	 * @param isSelected Is selected
+	 * @param hasFocus Has focus
+	 * @param row Row number
+	 * @param column Column number
+	 * @return JLabel
+	 */
+	private static Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
+		var component = new JLabel();
+		if (value != null) {
+			var hoColor = (HOColor) value;
+			Color color;
+			if (hoColor.getColorReference() != null) {
+				color = HOColor.getColor(hoColor.getHOColorName(), hoColor.getTheme());
+			} else {
+				color = hoColor.getColor();
+			}
+			component.setBackground(color);
+		}
+		return component;
+	}
+
+	/**
+	 * Create panels to edit theme selection and color settings
+	 */
+	private void initComponents() {
+		setLayout(new BorderLayout());
+		add(getTopPanel(), BorderLayout.NORTH);
+		add(getTablePanel(), BorderLayout.CENTER);
+	}
+
+	/**
+	 * Create panel with combo box for theme selection
+	 * @return Image Panel
+	 */
+	private JPanel getTopPanel() {
+		JPanel panel = new ImagePanel();
+		var themes = ThemeManager.instance().getRegisteredThemes();
+		var names = themes.stream().map(Theme::getName).toArray();
+		skins = new JComboBox(names);
+		var selected = ThemeManager.getCurrentThemeName();
+		skins.setSelectedItem(selected);
+		skins.addActionListener(this);
```

Comment:
These days I prefer to use a lambda here rather than making the panel an `ActionListener`, but up to you.

Reply:
good point - changed this.